ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionsOperation.java

```diff
@@ -116,4 +120,22 @@ private List<String> getPartitionNames(Table tbl) throws HiveException {
         desc.getOrder(), desc.getLimit());
     return partNames;
   }
+
+  private boolean isPartitionedIcebergTable(Table tbl) {
+    String t = tbl.getParameters().get("table_type");
+    return t != null && t.equalsIgnoreCase("ICEBERG") &&
+        (tbl.getStorageHandler().getPartitionTransformSpec(tbl).size() > 0);
+  }
+
+  private List<String> getIcebergParts(Table tbl) {
+    List<String> parts = new ArrayList<>();
+    try {
+      Table metaDataPartTable = context.getDb().getTable(tbl.getDbName(), tbl.getTableName(), "partitions", true);
+      parts = tbl.getStorageHandler().getIcebergPartitions( context.getConf(), metaDataPartTable);
+      return parts;
+    } catch (Exception e) {
```

Comment:
Do we have any more specific than a general Exception? And also, is that Warn message just a leftover?